sim/src/make/activity_model.rs

```diff
@@ -2,114 +2,264 @@ use crate::{
-        {
-            person.id = PersonID(s.people.len());
-            s.people.push(person);
-        }
+        };
+        let prob_local_worker = f64::min(
+            upper_bound_prob,
+            f64::max(lower_bound_prob, workers_cap as f64 / num_trips as f64),
+        );
 
-        // Create trips between map borders. For now, scale the number by the number of residences.
-        let incoming_connections = map.all_incoming_borders();
-        let outgoing_connections = map.all_outgoing_borders();
-        for mut person in timer
+        info!(
+            "BUILDINGS - workplaces: {}, residences: {}, mixed: {}",
+            prettyprint_usize(num_bldg_commercial),
+            prettyprint_usize(num_bldg_residential),
+            prettyprint_usize(num_bldg_mixed_residential_commercial)
+        );
+        info!(
+            "CAPACITY - workers_cap: {}, residents_cap: {}, prob_local_worker: {:.1}%, \
+             prob_local_resident: {:.1}%",
+            prettyprint_usize(workers_cap),
+            prettyprint_usize(residents_cap),
+            prob_local_worker * 100.,
+            prob_local_resident * 100.
+        );
+
+        let mut num_trips_local = 0;
+        let mut num_trips_commuting_in = 0;
+        let mut num_trips_commuting_out = 0;
+        let mut num_trips_passthru = 0;
+        timer.start("create people");
+
+        let all_outgoing_borders = map.all_outgoing_borders();
+        let mut tmp_rng = abstutil::fork_rng(rng);
+        let mut get_commuter_border = || {
+            for _ in 0..50 {
+                // TODO: prefer larger thoroughfares to better reflect reality.
+                let border = all_outgoing_borders.choose(&mut tmp_rng).unwrap();
+
+                // Only consider two-way intersections, so the agent can return the same way
+                // they came.
+                // TODO: instead, if it's not a two-way border, we should find an intersection
+                // an incoming border "near" the outgoing border, to allow a broader set of
+                // realistic options.
+                if border.is_incoming_border() {
+                    return TripEndpoint::Border(border.id, None);
+                }
+            }
+            debug_assert!(
+                false,
+                "failed to find a 2 way border in a reasonable time. Degenerate map?"
+            );
+            TripEndpoint::Border(all_outgoing_borders.choose(&mut tmp_rng).unwrap().id, None)
+        };
+
+        let person_params = (0..num_trips)
+            .map(|_| {
+                let (is_local_resident, is_local_worker) = (
+                    rng.gen_bool(prob_local_resident),
+                    rng.gen_bool(prob_local_worker),
+                );
+                let home = if is_local_resident {
+                    if let Some(residence) = residents.pop() {
+                        TripEndpoint::Bldg(residence)
+                    } else {
+                        warn!(
```

Comment:
The same warning could be spammed many times, right? For debugging, is it useful to see that? Not sure if it's worth complicating this by logging once and storing some state. (I'd almost want to add some method to Timer or similar to "log once and count occurrences", if we end up with lots of cases like this)

Reply:
Good point. 

Rather than logging each time we attempt to use unavailable capacity, I've changed it to log the capacity remaining at the end.